### ISSUE TITLE
[2][rrandom] Balance: 0 Displayed at Top When All of an Asset is in Open Order

### ISSUE THREAD

**Issue by landry314** [bitshares/bitshares-ui#206]
If an open order is holding ALL of an asset and I see "Unknown value" in the Equivalent Value column in Balances then the Balance calculation at the top on the bar says "Balance: 0". If I leave a little bit (truncate the amount to a whole number for example) before placing the order than the balance will calculate fine.

It is funny that this does NOT affect the TOTAL VALUE in the Accounts area under the [b] tab on the top left. There the balance is calculated perfectly fine.

I am using the Light Wallet on Mac OS X.

Does anyone else have this problem?

**Comment by wmbutler** [comment]
I'll leave this as a question until you can reproduce reliably.

**Comment by landry314** [comment]
I will, thanks. I'll even try to use the DevTools. Is there documentation on how to capture bugs with the DevTools? I would like to get the actual code for bugs like this.

**Comment by landry314** [comment]
#
![balance 0](https://user-images.githubusercontent.com/6866501/28608220-7cf5dec4-71ad-11e7-9f83-b1b9a106352c.jpg)

**Comment by landry314** [comment]
When every unit of an asset is in a open order then "Unknown value" shows up for me in the Equivalent Value column. I don't know why.

When there is nothing else in the Equivalent value column but unknown values then "Balance: 0" appears on the upper right.

The way I solve this is to truncate the order before placing it so there is a little bit of BTS left over in Balances then "Unknown value" does not appear and the Estimated Account Balance calculates correctly.

**Comment by landry314** [comment]
This is not a question, it is a bug. Did I reproduce it properly? Should I study it more to find out even more about how to reproduce it? For me it always happens when EVERY ASSET is zero OR completely in an open order causing its balance to be "Unknown Value". In essence, When the calculation consists only of "0" or "Unknown value" for every asset it thinks the balance is "0" ON THE TOP BAR. It is strange that in the "b" tab on the left under "Followed Accounts" the calculation is actually CORRECT! So why is the math different for balance calculation there and on the top bar???

**Comment by wmbutler** [comment]
Looks like this has been reproduced and documented well. Added it to the 170914 milestone.

**Comment by wmbutler** [comment]
Estimated at two hours if someone wants to claim this bounty.

**Comment by rrandom** [comment]
@wmbutler  Can I take this one? Thanks

**Comment by wmbutler** [comment]
It's yours. Adding it to 170901 if you think you can have it resolved by then.

**Comment by rrandom** [comment]
Hi, @landry314 , I don't understand where is ` the "b" tab on the left under "Followed accounts" ` with correct calculation.  Is there any screenshot?

**Comment by svk31** [comment]
This isn't currently the case, I already fixed it a few weeks ago so that it says 0 BTS in this case. We used to have the open order values displayed in brackets just after or below the 0 BTS value, but it quickly got very messy which is why the values are now split into "Balances" and "Open Orders".

Unless someone comes up with a better way of displaying the values I don't think there's anything to fix here.

**Comment by wmbutler** [comment]
Closing this as a wontfix.

**Comment by landry314** [comment]
@rrandom, the "b tab" is the bitshares logo on the far left, to the left of the "Accounts" tab. In there there WAS a section called "Followed accounts" but now it is just called "Accounts" further confusing the way accounts are managed. If you look at the "TOTAL VALUE" of the active account in the far right column, it DOES take into account the open orders when they say "Unknown value" as I described above while it says "Balance: 0" on the upper right.

If all your money is in an open order it should not say "Balance: 0" because that is not your account balance. I understand that you separated balances and open orders but the value on the upper right should be the TOTAL VALUE of the account and not just the total of the balances.

So, why not just change the section on the top to say "Value: XYX" instead of "Balance: XYZ" if that feels better for you. People expect that variable to represent the total value of their account and not to ignore open orders sometimes... or why not fix the "Unknown value" problem. Why would it have an unknown value? something is wrong there...

**Comment by btsfav** [comment]
TOTAL VALUE should be deleted to be honest. that's confusing many

**Comment by landry314** [comment]
@btsfav, how is it confusing? it is the total value of your account at the present which is a sum of balances and open orders while subtracting any debt. that is quite straightforward.

**Comment by btsfav** [comment]
@landry314 it's in BTS. this lead to many confused newbies, they sent in BTC for example, and thought it was converted to BTS. or people tried to send bitshares based on that number... the problems are endless

**Comment by landry314** [comment]
then change the unit in Settings -> General -> Preferred Unit of Account

I know, it is buried in there and would be nice if it were easier to change which is why I submitted issue #214.

**Comment by landry314** [comment]
Why not call it "ESTIMATED VALUE", that is more accurate anyway.

**Comment by landry314** [comment]
I honestly think this was closed as a wontfix due to confusion over what exactly this bug is. This could not possibly be an intentional action.

Anyway, I opened a new issue #374 to suggest a repurposing of this spot on the top bar to make it a value calculation instead of a balance calculation. This was really what it was intended to be anyway, it just got poorly implemented somewhere along the line.
